python/cudf/cudf/core/_base_index.py

```diff
@@ -1040,11 +1040,11 @@ def difference(self, other, sort=None):
         res_name = _get_result_name(self.name, other.name)
 
         if is_mixed_with_object_dtype(self, other):
-            difference = self.copy()
+            difference = self.copy().unique()
         else:
             other = other.copy(deep=False)
             difference = cudf.core.index._index_from_data(
-                cudf.DataFrame._from_data({"None": self._column})
+                cudf.DataFrame._from_data({"None": self._column.unique()})
                 .merge(
                     cudf.DataFrame._from_data({"None": other._column}),
```

Comment:
Should we also call `unique()` on the right hand side to make the merge smaller?

Reply:
I tried this with a few test cases, in some it had a better performance, and in some, it was worse. But pandas does this.

Reply:
OK, I'd just add a note here to alert future readers about potential performance issues:

```
# NOTE: may need to investigate calling `unique()` on the LHS before the merge for better performance 
```

Reply:
I am somewhat surprised that calling unique on the right column sometimes improves performance. `unique` calls `stable_distinct` which builds a hash table to uniquify things.

The `leftanti` join builds a hash table for the right column and then probes that hash table with the left column to return those rows in the left column that are _not_ in the hash table.

So calling `unique()` on the right column would just seem to be an extra hash-table build for no gain.

Can you show the test cases you ran to check performance @amiralimi ?